network/proc/pids_by_user.go

```diff
@@ -0,0 +1,82 @@
+package proc
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"syscall"
+
+	"github.com/safing/portbase/log"
+	"github.com/safing/portbase/utils"
+)
+
+var (
+	// pidsByUserLock is also used for locking the socketInfo.PID on all socket.*Info structs.
```

Comment:
As mentioned in my other comment, this is really really error prone as we are trying to protect a struct/interface that is not in the domain of the proc package.